### PR TITLE
Feat/4 cicd [FEAT] CI/CD 파이프라인 초기 구축 

### PR DESCRIPTION
**Pull request by 5hseok**
## #️⃣ Related Issue

Linked Feature backlog: #4 

## 📝 Purpose of PR

Jacoco와 SonarCloud를 사용한 CI/CD 파이프라인 구축 및 지속적인 코드 관리

## Contents

Jacoco와 SonarCloud 사용해서 DODAL_BE_SERVER에는 CI 작업을 실행하도록 했고, dnd 쪽 레포로 보내서 dnd 레포 main에 push 되면 CD 작업이 실행하도록 했습니다.  

다만, build.gradle애서의 Jacoco 커버지리 설정 및 스캔 범위와

```
// 브랜치 커버리지 최소 60%
            limit {
                counter = 'BRANCH'
                value = 'COVEREDRATIO'
                minimum = 0.60
            }

            // 라인 커버리지 최소한 80%
            limit {
                counter = 'LINE'
                value = 'COVEREDRATIO'
                minimum = 0.80
            }
...
//            // 커버리지 체크를 제외할 클래스들
//            excludes = [
//                    '**.*Application*',
//                    '**.*Request*',
//                    '**.*Response*',
//                    '**.*OAuthClient*',
//                    '**.*Interceptor*',
//                    '**.*Exception*',
//            ] + Qdomains
```

Jacoco와 SonarCloud가 정상적으로 이루어지는 지는 지속적인 테스트가 필요할 것 같습니다. 

## Checklist

-   [x] Does commit messages follow the convention?
-   [] Are variable names brief but descriptive?
-   [x] Is the code clean and easy to understand?
-   [ ] If a new feature has been added, or a bug fixed, has a test been added to confirm good behavior?
-   [ ] Does the test successfully test edge and corner cases?

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+# 의존성 파일 복사 및 다운로드
+COPY build.gradle .
+COPY settings.gradle .
+
+# 의존성 다운로드
+RUN ./gradlew dependencies
+
+# 소스코드 복사 및 빌드
+COPY ./ ./
```
Comment (sueelly) [inline Dockerfile:20]: `COPY * .`과 `COPY * ./`은 동일하지 않나요? 혹시 .과 ./ 둘 다 쓴 이유가 있나요?

소스코드 복사 시에 src/가 아니라 전체 복사인 이유가 있나요?

File: .github/workflows/OriginRepoCD.yml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+          else
+            echo "Skipping CI (Not a Forked Repo)"
+            exit 1
+          fi
+
+      # 1. 리포지토리 코드 체크아웃
+      - name: Check out the repository
+        uses: actions/checkout@v3
+
+#      # 2. properties 파일 설정 (JASYPT를 사용하므로 패스)
```
[the remark this answers is not in the document]
Comment (5hseok) [reply .github/workflows/OriginRepoCD.yml:29]: 넵넵 주석 제거해서 다시 올려놓겠습니다~

[the remark this answers is not in the document]
Comment (5hseok) [reply .github/workflows/OriginRepoCD.yml:29]: 그리고 OriginalRepoCD.yml과 Dockerfile에서 사용할 JASYPT_ENCRYPTOR_PASSWORD도 알려주시면 감사하겠습니다!

File: Dockerfile (hunk 1 above)
Comment (5hseok) [reply Dockerfile:20]: .과 ./은 같은 효과를 가지긴 합니다! 통일시키려는 생각은 해보지 못했네요!

Comment (5hseok) [reply Dockerfile:20]: 전체 복사를 한 이유는 추후에 빌드에 영향을 줄 수 있는 파일들이 추가되었을 경우를 대비하기 위해 진행했습니다! 위에서 build.gradle이나 gradlew, settings.gradle 등은 미리 복사해서 의존성 다운로드를 받아가긴 하지만 gradle이 아닌 다른 방식으로 파일이 추가될 경우를 생각했습니다.

Comment (5hseok) [reply Dockerfile:20]: 스캔 시간으로 인한 성능 저하보다 앞으로의 확장성을 생각해서 설정해두었다고 보시면 될 것 같습니다! 현재 단계에서는 큰 규모의 프로젝트가 아니라서 src를 스캔하는 것과 전체를 스캔하는 것의 차이가 미미하다고 생각해서요.

**Comment by 5hseok** [comment]
OriginRepoCD 부분 actions yml에서 properties 관련 주석 삭제해서 반영해두었습니다~

File: Dockerfile (hunk 1 above)
Comment (sueelly) [reply Dockerfile:20]: 오호 그러네요. 그런 경우를 생각지 못했네요. 덕분에 하나 배워갑니다~

File: .github/workflows/OriginRepoCD.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (sueelly) [reply .github/workflows/OriginRepoCD.yml]: 넵 DM으로 보내드리겠습니다~
